### PR TITLE
Fixed DIAMOND hsp normalization. I hope.

### LINKED ISSUE #436: DIAMOND output has qstart > qend when frame < 0
When DIAMOND blastx finds a match in a reverse complemented read, the qstart and qend numbers it emits are decreasing. I.e., they are indices into the read from its end, not its beginning. This is a difference from BLAST, which always has qstart < qend. Our DIAMOND hsp.py normalization code needs to take this into account.

### PR DESCRIPTION
**Pull request by terrycojones**
Added tests with all 6 frames.

Fixes #436.

### CODE REVIEW

File: dark/diamond/hsp.py [hunk 1]
```diff
@@ -1,150 +1,202 @@
...
-        # Note that readStartInSubject and readEndInSubject may not be 0 mod
-        # 3. They are offsets into the read string giving the position of
-        # the AA, which depends on the translation frame.
-        readStartInSubject = int(readStartInSubject / 3)
-        readEndInSubject = int(readEndInSubject / 3)
-
-    # No operations on original 1-based HSP variables (with underscores)
-    # should appear beyond this point.
+        # Convert queryLen and the query nucleotide start and end offsets
+        # to valid for the query after translation. When translating,
```
Comment (bamueh) [inline dark/diamond/hsp.py]: Shouldn't that say `...to valid X for the query...`?

File: dark/diamond/hsp.py [hunk 2]
```diff
@@ -1,150 +1,202 @@
...
+                    'queryEndInSubject (%d) < subjectEnd (%d)' %
+                    (queryEndInSubject, subjectEnd))
+
+
+def normalizeHSP(hsp, queryLen, diamondTask):
+    """
+    Examine an HSP and return information about where the query and the
+    alignment (match) begin and end.  Return a dict with keys that allow
+    the query and the alignment to be displayed relative to the subject
+    orientation (i.e., with start < stop for both the query and the
```
Comment (bamueh) [inline dark/diamond/hsp.py]: Can we use `match` rather than `alignment`? That would seem to be more consistent with the rest of our language.

**Comment by terrycojones** [comment]
Hmmm... the language in that comment is weird, as though it got
find&replace edited carelessly.  I'll change to match, though I think we
use alignment more consistently (e.g., in `alignments.py`). But match is a
good general word. I'll fix the blast one too. Thanks.

File: dark/diamond/hsp.py (hunk 1 above)
Comment (terrycojones) [inline dark/diamond/hsp.py]: Yes, fixed, thanks.

**Pull request merged**